subprojects/execution/src/test/groovy/org/gradle/internal/execution/steps/ResolveCachingStateStepTest.groovy

```diff
@@ -52,7 +52,7 @@ class ResolveCachingStateStepTest extends StepSpec<BeforeExecutionContext> {
         _ * context.validationProblems >> Optional.of({ ImmutableList.of("Validation problem") } as ValidationContext)
         1 * delegate.execute(work, { CachingContext context ->
             context.cachingState.disabledReasons*.category == [CachingDisabledReasonCategory.VALIDATION_FAILURE]
-            context.cachingState.disabledReasons*.message == ["Validation failed"]
+            context.cachingState.disabledReasons*.message == ["Caching has been disabled due to validation failures to ensure correctness. Please consult deprecation warnings for more details."]
```

Comment:
Suggestion:

```
Caching has been disabled to ensure correctness. Please consult deprecation warnings for more details.
```

As discussed, the term/concept of “validation failures” isn't meaningful to users an potentially confusing. I don't think it adds anything here and can be safely removed.